controller/service.go

```diff
@@ -104,6 +111,27 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 			c.clearServiceState(key, svc)
 			lbIPs = []net.IP{}
 		}
+
+		// If svc currently has 1 ip and policy PreferDualStack, try assigning ip from the missing family and same pool
+		if len(lbIPs) == 1 && familyPolicy == v1.IPFamilyPolicyPreferDualStack {
+			level.Info(l).Log("event", "tryAssignAdditionalIP", "msg", "familyPolicy is PreferDualStack, trying to assign additional ip")
+			existingFamily := ipfamily.ForAddress(lbIPs[0])
+			var additionalFamily ipfamily.Family
+			if existingFamily == ipfamily.IPv4 {
+				additionalFamily = ipfamily.IPv6
+			} else {
+				additionalFamily = ipfamily.IPv4
+			}
```

Comment:
> Thanks. I agree with your point, although:
> 
>     * Assign the value once instead of twice slightly increases the speed.
> 
>     * I think the project still uses `else` pretty frequently.
> 

We should try to avoid (and we are trying to enforce it as much as we can). If you are interested, I even made a talk about the recurring things we try to correct when reviewing https://archive.fosdem.org/2023/schedule/event/goreducecognitive/

Reply:
I'll check that out, thanks @fedepaol :smile:

Reply:
This was updated.